docs/proposals/PROPOSAL_C.md

```diff
@@ -0,0 +1,175 @@
+
+1. **Simple checking of key/value pairs**: A community might have an algorithm that is as simple as their tool checking for particular keys, and that values match for a host of interest.
+2. **Graph-based logic**: A community might warrant using advanced graph parsing to do a similar assessment. An example comes from [Archspec](https://github.com/archspec), where a single architecture (from a flat artifact or the platform, depending on what the tool chooses) can [be placed in a graph](https://github.com/archspec/archspec-json/blob/master/cpu/microarchitectures.json) and used for advanced assessment of compatibility.
+
+The above design encourages simplicity and designation of fewer metadata pieces in the OCI objects. Any complex graph-based query or additional relationships should be represented on the level of the tools that do the compatibility checking. This ensures that changes to these methods or graphs do not require re-creating extensive numbers of artifacts.
+
+## Modifications
+
+This proposal is agnostic to where the JSON object is placed. There are arguably pros and cons to different approaches, and the author (vsoch) leaves it to the expertise of the compatibility working group to decide on a "best" approach.
+
+### Image Manifest
+
+The JSON object could go directly in the image manifest or list to not require any additional artifact query or queries. For runtime tools, this would mean (still) just one call to retrieve all needed metadata to do a check. There would be no additional calls to retrieve plugins. Any compatibility checking plugins (discussed later in this proposal) would already be vetted and installed alongside the runtime tools.
+
+- **Pros**:
+  - Does not require additional query
+  - Cannot get separated from the image manifest or list
+  - Compatibility interest groups can develop artifacts that point TO the image (there is no required approval / permission to develop some niche compatibility definition).
+- **Cons**:
+  - Updating compatibility means also updating the manifest or list
+  - Compatibility interest groups need to advocate to get their metadata added (unlikely / challenging)
+
+### Compatibility Artifact
+
+If desired, a new **Compatibility Artifact** could be added (an OCI artifact) that stores this same JSON object as  a compatibility spec. The artifact would be, akin to other OCI artifacts, stored in a registry separate from an image manifest.
+
+- **Pros**:
+  - Empowers compatibility interest groups (even niche) to create artifacts for images
+- **Cons**:
+  - Can easily be separated from the original manifest / image (would require tooling to check for references)
+  - Requires additional queries
+
+### Image Manifest _and_ Compatibility Artifact
+
+We could get the best of both worlds by taking both approaches, and allowing the compatibility artifact to live in either place:
```

Comment:
Yeah, it was just a suggestion because I think we are talking about a common JSON object, but where you put it depends on the importance of it (atomic vs. community maintained). Perhaps we can find a balance, because each has pros and cons.